test/integration/nodebalancers_test.go

```diff
@@ -85,11 +85,15 @@ func setupNodeBalancer(t *testing.T, fixturesYaml string) (*linodego.Client, *li
 	t.Helper()
 	var fixtureTeardown func()
 	client, fixtureTeardown := createTestClient(t, fixturesYaml)
+	firewall, firewallTeardown, err := createFirewall(t, client)
+	defer firewallTeardown()
```

Comment:
The deferred function call will execute right after this function, and thus we probably don't want to do it in a setup function. Imagine that a test case call this setup function, and then the the teardown function will be executed right after the setup function, and then the test case won't be able to get the resource for testing. The recommended way is to put the `firewallTeardown` function call in the teardown function defined below (line 102-107).

You can also refer to this pattern in firewall test file. https://github.com/linode/linodego/blob/main/test/integration/firewalls_test.go#L161-L164